### PR TITLE
Add RBMutex.TryRLock

### PR DESCRIPTION
**Pull request by kkroo**
Attempt at adding `RBMutex.TryRLock` that tries to retrieve a reader lock token via the fast path or gives up returning nil.

### CODE REVIEW

File: rbmutex.go [hunk 1]
```diff
@@ -64,12 +64,19 @@ func NewRBMutex() *RBMutex {
...
+func (mu *RBMutex) TryRLock() *RToken {
+	if _, mux := mu.fastTryRlock(); mux != nil {
+		return mux
+	}
+	return nil
+}
+
+func (mu *RBMutex) TryLock() bool {
+	return mu.rw.TryLock()
+}
```
Comment (puzpuzpuz) [inline rbmutex.go:78]:
```suggestion
}

```

File: rbmutex.go [hunk 2]
```diff
@@ -64,12 +64,19 @@ func NewRBMutex() *RBMutex {
...
+// TryRLock tries to retrieve a reader lock token via
+// the fast path or gives up returning nil.
+func (mu *RBMutex) TryRLock() *RToken {
+	if _, mux := mu.fastTryRlock(); mux != nil {
+		return mux
+	}
+	return nil
+}
+
+func (mu *RBMutex) TryLock() bool {
```
Comment (puzpuzpuz) [inline rbmutex.go:76]: Godoc is missing here.

File: rbmutex.go [hunk 3]
```diff
@@ -64,12 +64,19 @@ func NewRBMutex() *RBMutex {
...
+	return mu.rw.TryLock()
+}
+func (mu *RBMutex) fastTryRlock() (*rslot, *RToken) {
```
Comment (puzpuzpuz) [inline rbmutex.go:79]: The return types could be `(int32, *RToken)`

**Review by puzpuzpuz — COMMENTED** [review]
The overall approach seems fine. Tests for the new methods are missing.

**Comment by kkroo** [comment]
nice library, thankz for the fast review.

**Comment by puzpuzpuz** [comment]
Thanks for the contribution! I'll add the missing tests separately.

**Pull request merged**